onyo/lib/onyo.py

```diff
@@ -260,6 +257,37 @@ def _generate_commit_message_subject(
         # return the shortest possible version of the commit message as fallback
         return msg
 
+    def sanitize_path(self, path: Union[Path, str]) -> Path:
+        """
+        Expects a relative or absolute path inside inside the repository.
+        These paths do not have to exist, just lead into the repo, so that the
+        function can be used to get paths to create new assets or move to
+        another location.
+
+        Returns an absolute `Path`.
+        """
+        onyo_path = Path(path).resolve()
+
+        # TODO: raise a more reasonable error
+        if not onyo_path.is_relative_to(self.root):
+            raise ValueError("path is not relative to root.")
```

Comment:
Minor suggestion, because `sanitize_paths` takes absolute paths, too. So, to error out with "not relative to ..." seems to suggest to a caller, that a relative path is required input, when really it's about not being part of the repo.

```suggestion
            raise ValueError(f"path {path} outside repository at {self.root}.")
```